app/lib/common/utilities/pdf_preview.dart

```diff
@@ -34,17 +35,38 @@ class _PdfPreviewState extends State<PdfPreview> {
             default:
               if (snapshot.hasError)
                 return Text('Could not build and save PDF: ${snapshot.error}');
-              else
+              else if (exit == true) {
+                //If PDF scaffold state changed switch to email_sender
+                return EmailRoute(pdf: snapshot.requireData);
+              } else
                 return PDFViewerScaffold(
                   appBar: AppBar(
                     title: Text('Example PDF'),
+                    actions: [
+                      OutlinedButton.icon(
+                          onPressed: () {
+                            exitPdf();
```

Comment:
I think this is anti-pattern, you should push the email PDF from navigator instead.

- [ ] Push the email widget into tree with like `Navigator.of(context).pushNamed...` because you have indeed added the `route` to routes already